File: sample/src/main/java/com/canzhang/sample/base/BaseManager.java
package com.canzhang.sample.base;


import android.app.Activity;
import android.app.Application;
import android.util.Log;
import android.widget.Toast;

import com.canzhang.sample.base.bean.ComponentItem;
import com.canzhang.sample.debug.DebugBaseApp;
import com.example.base.base.AppProxy;
import com.component.debugdialog.DebugDialog;

import java.util.List;

public abstract class BaseManager implements IManager {
    public Activity  mActivity;

    public BaseManager() {

    }

    public void showTipsDialog(String msg) {
        DebugDialog.getInstance().show("", msg);
    }

    public void showTipsDialog(String title, String msg) {
        DebugDialog.getInstance().show(title, msg);
    }

    public void log(String msg) {
        Log.e("Test", getClass().getSimpleName() + ":" + msg);
    }


    public void showToast(String msg) {
        Application application = AppProxy.getInstance().getApplication();
        if (application != null) {
            Toast.makeText(application, msg, Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public int getPriority() {
        return 0;
    }

    @Override
    public List<ComponentItem> getSampleItem(Activity activity) {
        mActivity = activity;
        return null;
    }
}

File: sample/src/main/java/com/canzhang/sample/base/BaseManager.kt
package com.canzhang.sample.base

import android.app.Activity
import android.util.Log
import android.widget.Toast
import com.canzhang.sample.base.bean.ComponentItem
import com.component.debugdialog.DebugDialog
import com.example.base.base.AppProxy

abstract class BaseManager : IManager {
    @JvmField
    var mActivity: Activity? = null
    fun showTipsDialog(msg: String?) {
        DebugDialog.getInstance().show("", msg)
    }

    fun showTipsDialog(title: String?, msg: String?) {
        DebugDialog.getInstance().show(title, msg)
    }

    fun log(msg: String) {
        Log.e("Test", javaClass.simpleName + ":" + msg)
    }

    fun showToast(msg: String?) {
        val application = AppProxy.getInstance().application
        if (application != null) {
            Toast.makeText(application, msg, Toast.LENGTH_SHORT).show()
        }
    }

    override fun getPriority(): Int {
        return 0
    }

    override fun getSampleItem(activity: Activity?): List<ComponentItem?>? {
        mActivity = activity
        return null
    }
}